cmd/internal/cli/verify.go

```diff
@@ -141,44 +136,64 @@ var VerifyCmd = &cobra.Command{
 }
 
 func doVerifyCmd(ctx context.Context, cmd *cobra.Command, cpath, url string) {
+	id, isGroup, err := ensureImageFlags(cmd, cpath, sifDescID, sifGroupID, verifyAll)
+	if err != nil {
+		sylog.Fatalf("%s", err)
+	}
+
+	// Dont need to resolve remote endpoint.
+	if !localVerify {
+		handleVerifyFlags(cmd)
+	}
+
+	author, _, err := signing.Verify(ctx, cpath, url, id, isGroup, verifyAll, authToken, localVerify, jsonVerify)
+	fmt.Printf("%s", author)
+	if err == signing.ErrVerificationFail {
+		sylog.Fatalf("Failed to verify: %s", cpath)
+	} else if err != nil {
+		sylog.Fatalf("Failed to verify: %s: %s", cpath, err)
+	}
+	sylog.Infof("Container verified: %s", cpath)
+}
+
+// ensureImageFlags takes a container path, sif id and group id, and formats to a more usable output.
+func ensureImageFlags(cmd *cobra.Command, cpath string, descrID, groupID uint32, all bool) (uint32, bool, error) {
```

Comment:
Can you comment a little more on the expected output here? It should e.g. be very obvious what the `all` boolean will return - if that's a special number that means 'everything'. Named returns would probably be helpful here too to clarify things.

Reply:
Yes, I will fix this function name, and comment to be more appropriate with what this function does. In this case, we pass the `all` here just to check to make sure that we are not using the `--sif-id`/`--group-id` flag at the same time.

Reply:
>  In this case, we pass the all here just to check to make sure that we are not using the --sif-id/--group-id flag at the same time.

this is the type of info which would be good in the comment really. The updated comment still doesn't really tell us what the function is/should be doing. Maybe something a bit more like...

```
// checkImageAndFlags verifies that the SIF image and SIF descriptor / group ID flags provided are usable
//  - checks that that a descrID and groupID are not being used together
//  - checks that a descrID or groupID are not being used with a request to verify all
// -  returns an error if flags are not usable
//  - returns an ID to verify, and true if it is a group ID / false if it is a descriptor ID
```

I think named returns would be good here to. Then in a nice editor we can easily see that the `bool` means `isGroup` from the function definition that pops up.